### PR TITLE
claim developer rewards router

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

**Review by alyn509 — DISMISSED** [review]

File: dex/router/src/factory.rs [hunk 1]
```diff
@@ -83,35 +84,23 @@ pub trait FactoryModule {
...
-                total_fee_percent,
-                special_fee_percent,
-                ManagedAddress::zero(),
-                MultiValueEncoded::new(),
-            )
-            .upgrade_from_source(
-                &self.pair_template_address().get(),
-                CodeMetadata::UPGRADEABLE | CodeMetadata::READABLE | CodeMetadata::PAYABLE_BY_SC,
-            );
+    fn upgrade_pair(&self, pair_address: ManagedAddress) {
```
Comment (psorinionut) [inline dex/router/src/factory.rs:91]: Why is this change needed? And in the claim developer rewards PR?

Comment (dorin-iancu) [reply dex/router/src/factory.rs:91]: It won't work with the current version of pair, it has no arguments for the upgrade function.

File: dex/pair/src/fee.rs [hunk 2]
```diff
@@ -418,5 +418,5 @@ pub trait FeeModule:
     fn trusted_swap_pair(&self) -> MapMapper<TokenPair<Self::Api>, ManagedAddress>;
 
     #[storage_mapper("whitelist")]
-    fn whitelist(&self) -> SetMapper<ManagedAddress>;
+    fn whitelist(&self) -> UnorderedSetMapper<ManagedAddress>;
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply dex/pair/src/fee.rs]: Fine. Reverted, but SetMapper is not needed here.

**Review by psorinionut — APPROVED** [review]

**Review by psorinionut — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]
